web3/contract.py

```diff
@@ -89,6 +89,56 @@
 ACCEPTABLE_EMPTY_STRINGS = ["0x", b"0x", "", b""]
 
 
+class ContractFunctions(object):
+    """Class containing contract function objects
+    """
+
+    _method_names = []
+
+    def __init__(self, contract):
+        self.contract = contract
+        self.function_methods = filter_by_type('function', self.contract.abi)
+        for method in self.function_methods:
+            self._method_names.append(method['name'])
+            setattr(self,
+                    method['name'],
+                    ContractMethod.factory(contract=self.contract,
+                                           web3=self.contract.web3,
+                                           method_name=method['name']))
+
+    def __getitem__(self, item):
+        if item in self._method_names:
+            return getattr(self, item)
+
+    def __call__(self):
+        return self
```

Comment:
This is odd.  It suggests that you could do `contract.functions().myMethod()` *and* `contract.functions.myMethod()`.  Unless there is a reason for the former, thoughts on removing the `__call__` and just sticking with property based access?